certora/harnesses/ERC20WrapperHarness.sol

```diff
@@ -2,10 +2,15 @@
 
 pragma solidity ^0.8.20;
 
-import "../patched/token/ERC20/extensions/ERC20Wrapper.sol";
+import {ERC20Permit} from "../patched/token/ERC20/extensions/ERC20Permit.sol";
```

Comment:
Do we need to add ERC20Permit in this harness ?

Reply:
It's needed because the `ERC20Wrapper.spec` imports the `ERC20.spec`, which imports `IERC2612.spec`. Although that spec is not run for the wrapper, the functions declared for `IERC2612` are expected there. The easiest workaround I found was to add permit to the harness.

Is there any alternative you can think of?

Reply:
My understanding was that it was ok to import an interface that the target contract does not implement. Any rule that includes a call to these function would be skipped. In this case that means that running the ERC20.spec on the ERC20Wrapper would not check all of the properties, `rule permit` and `rule onlyHolderOfSpenderCanChangeAllowance` would be skipped.

Did that change in CVL2?

Reply:
Yes, it changed and now the alternative is to add an `optional` annotation. More [here](https://docs.certora.com/en/latest/docs/cvl/cvl2/changes.html?highlight=bytes32#optional-methods-block-entries).

The issue is not because the rules may include calls to functions that aren't implemented, but because the `envfreeFuncsStaticCheck`. Previously in CVL1 any unknown method was just skipped but since CVL2 they're indeed checked.

Adding the `optional` annotation is an alternative but in that case we may accidentally skip it in the `ERC20.spec` as well.

Reply:
Good to know !

So its between that and `optional`. Either is fine I guess.

The `optional` would possibly be better if we think users will use our specs to validate their code (that may not be Permit). But unless that is something we consider I don't think we need to change the spec.